rest-service/manager_rest/test/endpoints/test_deployment_labels_dependencies.py

```diff
@@ -1,613 +1,420 @@
-        self.client.deployments.update_labels('dep_0', [
-                {
-                    'csys-obj-parent': 'dep_1'
-                }
+    def test_sub_deployments_counts_after_convert_to_service(self):
+        env1 = self._deployment(id='env1')
+        self._deployment(id='env2')
+        self._deployment(id='srv1')
+        self.client.deployments.update_labels(
+            'env2',
+            labels=[
+                {'csys-obj-type': 'environment'},
+                {'csys-obj-parent': 'env1'}
             ]
         )
-
-        self.put_deployment(deployment_id='dep_11', blueprint_id='dep_11')
-        self.put_deployment(deployment_id='dep_12', blueprint_id='dep_12')
-        self.put_deployment(deployment_id='dep_13', blueprint_id='dep_13')
-        self.put_deployment(deployment_id='dep_14', blueprint_id='dep_14')
-
-        self.client.deployments.update_labels('dep_1', [
-                {
-                    'csys-obj-parent': 'dep_11'
-                }
+        self.client.deployments.update_labels(
+            'srv1',
+            labels=[
+                {'csys-obj-parent': 'env1'}
             ]
         )
+        assert env1.sub_services_count == 1
+        assert env1.sub_environments_count == 1
 
-        self.client.deployments.update_labels('dep_11', [
-                {
-                    'csys-obj-parent': 'dep_12'
-                }
+        self.client.deployments.update_labels(
+            'env2',
+            labels=[
+                {'csys-obj-type': 'service'},
+                {'csys-obj-parent': 'env1'}
             ]
         )
+        assert env1.sub_services_count == 2
+        assert env1.sub_environments_count == 0
 
-        self.client.deployments.update_labels('dep_12', [
-                {
-                    'csys-obj-parent': 'dep_13'
-                }
+    def test_sub_deployments_counts_after_convert_to_environment(self):
+        env1 = self._deployment(id='env1')
+        self._deployment(id='env2')
+        self._deployment(id='srv1')
+        self.client.deployments.update_labels(
+            'env2',
+            labels=[
+                {'csys-obj-type': 'environment'},
+                {'csys-obj-parent': 'env1'}
             ]
         )
-
-        self.client.deployments.update_labels('dep_13', [
-                {
-                    'csys-obj-parent': 'dep_14'
-                }
+        self.client.deployments.update_labels(
+            'srv1',
+            labels=[
+                {'csys-obj-parent': 'env1'}
             ]
         )
-        dep_graph = RecursiveDeploymentLabelsDependencies(self.sm)
-        dep_graph.create_dependencies_graph()
-        targets = dep_graph.find_recursive_deployments(['dep_0'])
-        self.assertEqual(len(targets), 5)
-        self.assertIn('dep_1', targets)
-        self.assertIn('dep_11', targets)
-        self.assertIn('dep_12', targets)
-        self.assertIn('dep_13', targets)
-        self.assertIn('dep_14', targets)
```

Comment:
tests for `RecursiveDeploymentLabelsDependencies` ie. the graph, need to go: the graph is going to be unused very soon, and I don't want to keep them around.

Reply:
also the recently-added `recalc_ancestors` tests, cover checking the "graph"/"tree" in more depth than those